### ISSUE TITLE
jenkins: tests extremely slow on Solaris and some

### ISSUE THREAD

**Issue by hzhou** [pmodels/mpich#4548]
Our Jenkins Solaris node has exact the same hardware as the centos nodes, however, the tests are running nearly 5 times slower on Solaris than centos. It has always been slower, but lately seems it is getting worse. Need investigate.

Separately, `ch4-ofi` with intel compiler and external `libfabric` runs up to 8 hours. Typically `ch4-ofi` finishes tests in a hour.

**Comment by wesbland** [comment]
It's very strange that ch4/ofi would run for hours with ICC and an external libfabric. We build it that way now and a job is done in 1:45 (not sure if that was scheduled immediately or had to wait in the Slurm queue).

**Comment by hzhou** [comment]
> It's very strange that ch4/ofi would run for hours with ICC and an external libfabric. We build it that way now and a job is done in 1:45 (not sure if that was scheduled immediately or had to wait in the Slurm queue).

The slow probably is coming from the compiling -- although why it is slow there needs investigate.

Anyway, I just realized how critical `make all` in testing is important. I always thought the tests are small and takes negligible time to make compared to the actual testing. I was very long. Turns out, some of the test is very slow to compile and it is duplicately used in many tests. In particular, `coll/allred` takes like 15 seconds to compile and it is being used in 46 tests, thanks to the collective cvar tests. So we were like spending like more than 10 minutes just on compiling that test repeated alone! 

We are shifting to `make all` before `make testing` now. I know you guys have been doing this for a long time. You listen, but you never here, until you see the problem yourself 😄 

PS: We switched to `make all` yesterday, and the runtime for Intel compiler, external libfabric, ch4:ofi was normal now.
